src/coreclr/jit/fgehopt.cpp

```diff
@@ -2437,7 +2437,18 @@ PhaseStatus Compiler::fgTailMergeThrows()
                 case BBJ_SWITCH:
                 {
                     JITDUMP("*** " FMT_BB " now branching to " FMT_BB "\n", predBlock->bbNum, canonicalBlock->bbNum);
+                    FlowEdge* prevEdge      = fgGetPredForBlock(nonCanonicalBlock, predBlock);
+                    weight_t  removedWeight = prevEdge->getLikelyWeight();
                     fgReplaceJumpTarget(predBlock, nonCanonicalBlock, canonicalBlock);
+
+                    // In practice, when we have true profile data, we can repair it locally here, since the no-return
+                    // call means that there is no contribution from nonCanonicalBlock to any of its successors.
+                    // Note that this might not be the case if we have profile data from e.g. synthesis, so this
+                    // repair is best-effort only.
+                    if (canonicalBlock->hasProfileWeight())
+                    {
+                        canonicalBlock->setBBProfileWeight(canonicalBlock->bbWeight + removedWeight);
```

Comment:
If you enable profile consistency checks for throw merging, I suspect this will cause asserts; since we don't convert `canonicalBlock` into a `BBJ_THROW` block until morph, the post-phase checks will see new weight into `canonicalBlock` that isn't being propagated to its successors. If you add some logic to make `fgPgoConsistent` false here, I think you'll be able to enable profile checks after this phase without issue (though I'm ok with you leaving them off for now, I can enable them in another PR).

Also, sorry for creating merge conflicts here

Reply:
Right, the comment above explains the situation. In practice this local update keeps global consistency, but in synthetic cases it does not. That's why I kept the profile checking disabled for this phase. Do you prefer to move the disablement further in favor of setting `fgPgoConsistent = false` here? If so I'll do that.

Reply:
Up to you -- I have the checks enabled in a draft PR (#111047), but I'm ok with rebasing that on top of this one

Reply:
Let's just close this one in favor of #111047